src/Options.php

```diff
@@ -558,7 +577,40 @@ public function setIntegrations(array $integrations): void
      */
     public function getIntegrations(): array
     {
-        return $this->options['integrations'];
+        $defaultIntegrations = $this->options['default_integrations'] ? $this->defaultIntegrations : [];
+        $userIntegrations = $this->options['integrations'];
+        $integrations = [];
+
+        if (\is_array($userIntegrations)) {
+            $userIntegrationsClasses = array_map('get_class', $userIntegrations);
+
+            /** @var array<string, bool> $pickedIntegrationsClasses */
+            $pickedIntegrationsClasses = [];
+
+            foreach ($defaultIntegrations as $defaultIntegration) {
+                $defaultIntegrationClass = \get_class($defaultIntegration);
+
+                if (!\in_array($defaultIntegrationClass, $userIntegrationsClasses, true)) {
+                    $pickedIntegrationsClasses[$defaultIntegrationClass] = true;
+                    $integrations[] = $defaultIntegration;
+                }
+            }
+
+            foreach ($userIntegrations as $userIntegration) {
+                $userIntegrationClass = \get_class($userIntegration);
+
+                if (isset($pickedIntegrationsClasses[$userIntegrationClass])) {
+                    continue;
+                }
+
+                $pickedIntegrationsClasses[$userIntegrationClass] = true;
+                $integrations[] = $userIntegration;
+            }
```

Comment:
What is this filter about? I'm not sure I understand this logic, but it seems complicated and founded on some assumption that the end user could miss.

Reply:
The logic is a bit complicated (I carried it over from the JS SDK) and what it does is:

- It ensures that just one instance of the same integratione exists in the list
- It adds all the missing default integrations if the `enable_integrations` option is set to `true`
- It calls the user-supplied callable by passing the list of default integrations if any

Reply:
I don't like that you're prioritizing the default integrations over the user supplied ones. An user could want to change the behavior of a default integration by supplying it's own, maybe changing just the constructor arguments, and in this way you're losing that.

Also, I think that you could simplify this logic by using an array with FQCN as keys, and do an array merge after.